### PR TITLE
Add native support for 2D images + fix indexing bug in a GPU kernel

### LINKED ISSUE #34: Add native support for 2D images
Currently, a dummy third axis needs to be added to 2D images, otherwise it breaks the code.

### LINKED ISSUE #37: `interpolate_edge_values` breaks when a GPU array is passed
We're using scalar indexing and GPUs don't like that (b/c it makes them slow!)

### PR DESCRIPTION
**Pull request by ma-sadeghi**
Fixes #37
Closes  #34

## Summary by CodeRabbit

- **New Features**
  - Added a function to ensure arrays have at least three dimensions for improved data handling.
  - Extended test suite with new 2D test cases verifying tortuosity and formation factors.

- **Refactor**
  - Improved performance and readability by vectorizing edge value interpolation in simulations.
  - Enhanced simulation setup with better preprocessing, device management, and informative logging.

- **Style**
  - Updated script headers and comments for improved clarity.

- **Chores**
  - Reduced default image size in the demo script for faster execution.

### CODE REVIEW

File: src/numpytools.jl [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+    if ndims(x) == 0
+        reshape([x], 1, 1, 1)
+    elseif ndims(x) == 1
+        reshape(x, length(x), 1, 1)
+    elseif ndims(x) == 2
+        reshape(x, size(x)..., 1)
+    else
+        x  # Already 3D or higher
+    end
+end
```
[the remark this answers is not in the document]
Comment (ma-sadeghi) [reply src/numpytools.jl:11]: Hmm, I think array type is preserved, see `[x]`, we're literally using the same scalar. Sure, if we meant `x` to be on GPU, then it breaks, but scalars are always on CPU anyways, so there's no way to intercept that. Besides, scalars are not meant to be passed at least practically. `img` should be an image, even 1D arrays are not even practically defined.
